src/renderer/js/component/shapeShift/internal/active-shift.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { PureComponent } from "react";
+import { shell, clipboard } from "electron";
+import QRCode from "qrcode-react";
+import * as statuses from "constants/shape_shift";
+import { Address } from "component/common";
+import Link from "component/link";
+
+export default class ActiveShapeShift extends PureComponent {
+  constructor() {
+    super();
+    this.continousFetch = undefined;
+  }
+
+  componentDidMount() {
+    const { getActiveShift, shiftDepositAddress } = this.props;
+
+    getActiveShift(shiftDepositAddress);
+    this.continousFetch = setInterval(() => {
+      getActiveShift(shiftDepositAddress);
+    }, 10000);
+  }
+
+  componentWillUpdate(nextProps) {
+    const { shiftState } = nextProps;
+    if (shiftState === statuses.COMPLETE) {
+      this.clearContinuousFetch();
+    }
+  }
+
+  componentWillUnmount() {
+    this.clearContinuousFetch();
+  }
+
+  clearContinuousFetch() {
+    clearInterval(this.continousFetch);
+    this.continousFetch = null;
+  }
+
+  render() {
+    const {
+      shiftCoinType,
+      shiftDepositAddress,
+      shiftReturnAddress,
+      shiftOrderId,
+      shiftState,
+      shiftDepositLimit,
+      clearShapeShift,
+      doShowSnackBar,
+    } = this.props;
+
+    return (
+      <div className="shapeshift__active-shift">
+        {shiftState === statuses.NO_DEPOSITS && (
+          <div>
+            <p>
+              Send up to{" "}
+              <span className="font-bold">
+                {shiftDepositLimit}{" "}
+                <span className="shapeshift__success">{shiftCoinType}</span>
+              </span>{" "}
+              to the address below
+            </p>
+
+            <div className="shapeshift__deposit-address-wrapper">
+              <Address address={shiftDepositAddress} />
+              {/* the header__item class should be a generic button class */}
+              <span className="header__item">
+                <Link
```

Comment:
Can this behavior be added to the existing `Address` component? If not all `<Address>` components should have a button, then this can be an optional property.

Reply:
Good idea